resources/syntax/Twig.sublime-syntax

```diff
@@ -207,8 +207,9 @@ contexts:
 ###[ Filters ]#################################################################
 
   filter:
-    - match: \|
-      scope: keyword.operator.pipe.twig
+    - match: \s*(\|)\s*
+      captures:
+        1: keyword.operator.pipe.twig
```

Comment:
How would I prevent whitespace after the last filter name from being included so that test like [this one](https://github.com/Sublime-Instincts/BetterTwig/blob/506db5fc3a6008679a15297927c04102388fd6fb/resources/syntax/tests/syntax_test_html.twig#L57) pass?

Reply:
Assuming function arguments directly follow function names without spaces allowed, a possible solution would be to modify `functions_arguments` context


```yml
  functions_arguments:
    - match: \(
      scope: punctuation.section.group.begin.twig
      set: functions_arguments_body   # use set instead of push
    - include: immediately_pop   # added as bailout

  functions_arguments_body:
    - meta_scope: meta.function-call.arguments.twig meta.group.twig
    - match: \)
      scope: punctuation.section.group.end.twig
      pop: 1  # reduce contexts' being popped
```

and than adjusting filter to

```yml
  filter:
    - match: \|
      scope: keyword.operator.pipe.twig
      push: filter_body

  filter_body:
    - meta_scope: meta.function.filter.twig
      # builtin filters
    - match: '{{builtin_filters}}'
      scope: meta.function-call.identifier.twig support.function.filter.twig
      set: function_arguments
      # custom filters
    - match: '{{identifiers}}'
      scope: meta.function-call.identifier.twig variable.function.filter.twig
      set: function_arguments
    - include: else_pop

```

Reply:
Thanks for the tip, I'm going to try to rework it. I took a run at it with what you suggested, but it loses the meta scope that way, so I'll have to figure out a way to retain it.

Reply:
Appears Twig requires various adjustments to align with other template syntaxes.

I already started to apply some of them in an interal WIP as it is a relatively easy task with having reworked all the others as well.

Hope I can make some progress in the next few days.

Maybe we could work on that new base.

Reply:
That would be great, it would make some of these changes a lot easier. I'll hold for now so I can rebase afterwards.